### PR TITLE
Add option to ignore specific property

### PR DESCRIPTION
**Pull request by michealmouner**
## Subject

I am targeting this branch, because I need to add t an option for ignoring a property 
now both will work
```
global_ignore_columns: # old one still working
    - created_at
    - updated_at

entity_ignored_properties: # new one
    App\Entity\EntityOne:
        - propertyOne
        - propertyTwo
    App\Entity\EntityTwo:
        - propertyOne
```

## Changelog

```markdown
### Added
two new options:
- entity_ignored_properties --> allow to ignore property for specific entity

### Fixed

```

### CODE REVIEW

**Review by VincentLanglet — DISMISSED** [review]

**Comment by VincentLanglet** [comment]
Can you take a look @phansys ?

File: src/EventListener/LogRevisionsListener.php [hunk 1]
```diff
@@ -411,6 +431,12 @@ private function saveRevisionEntityData($class, $entityData, $revType): void
                 continue;
             }
 
+            if ($class->isIdentifier($field) && !empty($entityData[$field]) && is_scalar($entityData[$field])) {
+                $params[] = $entityData[$field];
+                $types[] = $class->getTypeOfField($field);
+                continue;
```
Comment (phansys) [inline src/EventListener/LogRevisionsListener.php:437]:
```suggestion

                continue;
```

File: tests/BaseTest.php [hunk 2]
```diff
@@ -178,7 +178,8 @@ protected function getAuditManager(): AuditManager
         }
 
         $auditConfig = AuditConfiguration::forEntities($this->auditedEntities);
-        $auditConfig->setGlobalIgnoreColumns(['ignoreme']);
+        $auditConfig->setGlobalIgnoreColumns(['ignoreMe']);
+        $auditConfig->setEntityIgnoredProperties(['SimpleThings\EntityAudit\Tests\Fixtures\Core\ProfileAudit' => ['ignoreProperty']]);
```
Comment (phansys) [inline tests/BaseTest.php:182]: Could we use the `class` pseudo-constant for the `\SimpleThings\EntityAudit\Tests\Fixtures\Core\ProfileAudit` class?

File: src/AuditReader.php [hunk 3]
```diff
@@ -828,7 +840,8 @@ private function createEntity($className, array $columnMap, array $data, $revisi
         $this->entityCache[$className][$key][$revision] = $entity;
 
         foreach ($data as $field => $value) {
-            if (isset($class->fieldMappings[$field])) {
+            $isIgnoredProperty = isset($columnMap[$field]) && $this->config->isEntityIgnoredProperty($class->getName(), $field);
```
Comment (phansys) [inline src/AuditReader.php:843]: We should avoid one-time variables.

File: src/AuditConfiguration.php [hunk 4]
```diff
@@ -172,4 +173,25 @@ public function getRevisionIdFieldType()
     {
         return $this->revisionIdFieldType;
     }
+
+    /**
+     * @return array<string, string[]>
+     */
+    final public function getEntityIgnoredProperties(): array
```
Comment (phansys) [inline src/AuditConfiguration.php:180]: Maybe we could add the model name as argument 1 in order to allow this method to check directly against a specific combination of model and property.

**Pull request closed without merging**